d2/src/main/java/com/linkedin/d2/discovery/stores/zk/ZooKeeperEphemeralStore.java

```diff
@@ -227,21 +234,19 @@ public void onSuccess(None none)
         }
         final String ephemeralPath = path + "/" + ephemeralPrefix + "-";
 
-        AsyncCallback.StringCallback stringCallback = new AsyncCallback.StringCallback()
-        {
-          @Override
-          public void processResult(int rc, String path, Object ctx, String name)
+        AsyncCallback.StringCallback stringCallback = (rc, path1, ctx, name) -> {
+          KeeperException.Code code = KeeperException.Code.get(rc);
+          switch (code)
           {
-            KeeperException.Code code = KeeperException.Code.get(rc);
-            switch (code)
-            {
-              case OK:
-                callback.onSuccess(None.none());
-                break;
-              default:
-                callback.onError(KeeperException.create(code));
-                break;
-            }
+            case OK:
+              notifyZnodePathAndDataCallback(prop, name, value.toString()); // set the created znode path, such as "/d2/uris/ClusterA/hostA-1234"
```

Comment:
Should this be `ephemeralPath` instead of `name`?

Reply:
nope, name is the one with the sequential number suffix created by Zookeeper. (so ephemeralPath only has lor1.xxx, but name has lor1.xxx-123456)